Sources/ClientRuntime/Networking/Http/CRT/CRTClientEngine.swift

```diff
@@ -84,30 +84,72 @@ public class CRTClientEngine: HttpClientEngine {
-            } catch {
-                continuation.resume(throwing: error)
+        switch connection.httpVersion {
+        case .version_1_1:
+            self.logger.debug("Using HTTP/1.1 connection")
+            let crtRequest = try request.toHttpRequest()
+            return try await withCheckedThrowingContinuation { (continuation: StreamContinuation) in
+                let requestOptions = makeHttpRequestStreamOptions(request: crtRequest,
+                                                                  continuation: continuation)
+                do {
+                    let stream = try connection.makeRequest(requestOptions: requestOptions)
+                    try stream.activate()
+                } catch {
+                    continuation.resume(throwing: error)
+                }
+            }
+        case .version_2:
+            self.logger.debug("Using HTTP/2 connection")
+            let crtRequest = try request.toHttp2Request()
+            return try await withCheckedThrowingContinuation { (continuation: StreamContinuation) in
+                let requestOptions = makeHttpRequestStreamOptions(request: crtRequest,
+                                                                  continuation: continuation,
+                                                                  http2ManualDataWrites: true)
+                let stream: HTTP2Stream
+                do {
+                    stream = try connection.makeRequest(requestOptions: requestOptions) as! HTTP2Stream
+                    try stream.activate()
+                } catch {
+                    continuation.resume(throwing: error)
+                    return
+                }
+
+                // At this point, continuation is resumed when the initial headers are received
+                // it is now safe to write the body
+                // writing is done in a separate task to avoid blocking the continuation
+                Task {
+                    do {
+                        try await stream.write(body: request.body)
+                    } catch {
+                        self.logger.error(error.localizedDescription)
+                    }
+                }
             }
-        })
+        case .unknown:
+            fatalError("Unknown HTTP version")
+        }
     }
 
-    public func makeHttpRequestStreamOptions(
-        _ request: SdkHttpRequest,
-        _ continuation: StreamContinuation
-    ) throws -> HTTPRequestOptions {
+    /// Creates a `HTTPRequestOptions` object that can be used to make a HTTP request
+    /// - Parameters:
+    ///   - request: The `HTTPRequestBase` object that contains the request information
+    ///   - continuation: The continuation that will be resumed when the request is complete
+    ///   - http2ManualDataWrites: Whether or not the request is using HTTP/2 manual data writes, defaults to `false`
+    ///     If not set to true, HTTP/2 manual data writes will be disabled and result in a runtime error.
```

Comment:
```suggestion
    ///     If set to false, HTTP/2 manual data writes will be disabled and result in a runtime error.
```

Can you provide more information in what else is a factor here in causing the runtime error. If this flag defaults to false, reading the comment makes it sound like this function will always result in a runtime error then...